x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/TokenService.java

```diff
@@ -2044,8 +2044,8 @@ private void initialize(ClusterService clusterService) {
     // to prevent too many cluster state update tasks to be queued for doing the same update
     private final AtomicBoolean installTokenMetadataInProgress = new AtomicBoolean(false);
 
-    private void installTokenMetadata(MetaData metaData) {
-        if (metaData.custom(TokenMetaData.TYPE) == null) {
+    private void installTokenMetadata(ClusterState state) {
+        if (state.custom(TokenMetaData.TYPE) == null) {
```

Comment:
This one does not really contribute to the slowdown, but I found this one when I was investigating things and was repeatedly seeing  `install-token-metadata` cluster state updates (which become NOOPs) because the condition here is checking the wrong customs. /cc: @jkakavas